### PR TITLE
add load optimmethod for python

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

fix issue https://github.com/intel-analytics/BigDL/issues/589
add load optimmethod for python

### CODE REVIEW

File: pyspark/bigdl/optim/optimizer.py [hunk 1]
```diff
@@ -313,6 +313,26 @@ def __init__(self,
         JavaValue.__init__(self, None, bigdl_type, monitor, factor, patience, mode, epsilon,
                            cooldown, min_lr)
 
+class OptimMethod(JavaValue):
```
Comment (zhichao-li) [inline pyspark/bigdl/optim/optimizer.py]: This should be the base class of SGD and anyother concrete OptimMethod.

File: pyspark/bigdl/optim/optimizer.py [hunk 2]
```diff
@@ -313,6 +313,26 @@ def __init__(self,
...
+    def __init__(self, bigdl_type="float"):
+        self.bigdl_type = bigdl_type
+
+    def load(self, path):
```
Comment (zhichao-li) [inline pyspark/bigdl/optim/optimizer.py]: add static annotation to turn this be a static method.

File: pyspark/bigdl/optim/optimizer.py [hunk 3]
```diff
@@ -313,6 +313,26 @@ def __init__(self,
...
+        """
+        load optim method
+        :param path: file path
+        """
+        return callBigDlFunc(self.bigdl_type, "createLoadOptimMethod", path)
+
+    def save(self, method, path, overWrite):
```
Comment (zhichao-li) [inline pyspark/bigdl/optim/optimizer.py]: remove parameter method: `method`, self should be the method.

File: pyspark/bigdl/optim/optimizer.py [hunk 4]
```diff
@@ -313,6 +313,26 @@ def __init__(self,
...
+        """
+        save OptimMethod
+        :param method    method to be saved
+        :param path      path
+        :param overWrite whether to overwrite
+        """
+        return callBigDlFunc(self.bigdl_type, "createSaveOptimMethod", method, path, overWrite)
```
Comment (zhichao-li) [inline pyspark/bigdl/optim/optimizer.py]: Rename `createSaveOptimMethod` to `saveOptimMethod`

File: pyspark/bigdl/optim/optimizer.py [hunk 5]
```diff
@@ -313,6 +313,26 @@ def __init__(self,
...
+class OptimMethod(JavaValue):
+    def __init__(self, bigdl_type="float"):
+        self.bigdl_type = bigdl_type
+
+    def load(self, path):
+        """
+        load optim method
+        :param path: file path
+        """
+        return callBigDlFunc(self.bigdl_type, "createLoadOptimMethod", path)
```
Comment (zhichao-li) [inline pyspark/bigdl/optim/optimizer.py]: `createLoadOptimMethod` to `loadOptimMethod`

File: pyspark/test/simple_integration_test.py [hunk 6]
```diff
@@ -72,6 +72,39 @@ def test_load_model(self):
...
+        model.add(l1)
+
+        sgd = SGD(learningrate=0.01, learningrate_decay=0.0002, weightdecay=0.0,
+                  momentum=0.0, dampening=0.0, nesterov=False,
+                  leaningrate_schedule=Poly(0.5, int((data_len/batch_size)*epoch_num)))
+
+        tmp_path = tempfile.mktemp()
+        optim = OptimMethod()
+        optim.save(sgd, tmp_path, True)
+        optim_method = optim.load(tmp_path)
```
Comment (zhichao-li) [inline pyspark/test/simple_integration_test.py]: should we check part of the re-loaded fields is same as the original one or not?

**Comment by zhichao-li** [comment]
Seems like there's no `resume` api in Scala, maybe we need to add one for scala as well. @yiheng

**Comment by zhangxiaoli73** [comment]
jekins pass

**Comment by zhangxiaoli73** [comment]
@shane-huang do you think we need to add another resume api in bigdl ? Now, user can load model and optim-method to resume training.

**Comment by zhichao-li** [comment]
LGTM

**Comment by shane-huang** [comment]
I don't think we need a resume API.  Keras doesn't have one either.  It is good we allow OptimMethod and Model to be loaded seperately. Sometimes we only need to load the model without the OptimMethod.

**Comment by shane-huang** [comment]
Please change the docs accordingly. https://bigdl-project.github.io/master/#APIdocs/Optimizers/OptimMethod/ 

Also in https://bigdl-project.github.io/master/#ProgrammingGuide/optimization/#resume-training
change ``` (Scala only, Python does not support this yet) ```
> Use Module.load (Scala) or Model.load(Python) to load the model snapshot into an model object, and OptimMethod.load (Scala only, Python does not support this yet) to load optimization method into an OptimMethod object.

**Comment by zhangxiaoli73** [comment]
update md

**Pull request merged**